lib/internal/Magento/Framework/Logger/Handler/Base.php

```diff
@@ -34,11 +34,15 @@ class Base extends StreamHandler
      */
     public function __construct(
         DriverInterface $filesystem,
-        $filePath = null
+        $filePath = null,
+        $fileName = null
     ) {
         $this->filesystem = $filesystem;
+        if(null === $fileName){
+            $fileName = $this->fileName;
```

Comment:
I think it should be the other way around: `if (!empty($filename)) { $this->fileName = $fileName }`. If I pass a new file path to the class, I'm expecting its state to change, as well, but you're not changing `$this->fileName` at all.